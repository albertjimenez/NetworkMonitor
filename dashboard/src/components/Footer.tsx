import React from "react";
import {getCurrentYear} from "../utils/dateUtils.ts";
import {Layout} from "antd";


const Footer: React.FC = () => (
    <Layout.Footer style={{textAlign: 'center'}}>
        &copy; {getCurrentYear()} Network Monitor. All rights reserved.
    </Layout.Footer>
);
export default Footer;
