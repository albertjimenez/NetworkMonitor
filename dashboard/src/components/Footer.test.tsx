import { render, screen } from '@testing-library/react';
import Footer from './Footer';
import {describe, expect, it, vi} from 'vitest';
import '@testing-library/jest-dom';  // Import this to use toBeInTheDocument()

// Mock the getCurrentYear function
vi.mock('../utils/dateUtils.ts', () => ({
    getCurrentYear: () => 2024, // Mock it to return a fixed year
}));

describe('Footer component', () => {
    it('renders the footer with the correct year and text', () => {
        render(<Footer/>);

        // Check that the footer text includes the mocked current year
        const footerText = screen.getByText(/Â© 2024 Network Monitor. All rights reserved./i);
        expect(footerText).toBeInTheDocument();
    });
    it('Footer matches snapshot', () => {
        const footerComponent = render(<Footer/>);
        expect(footerComponent).toMatchSnapshot();
    });
});
