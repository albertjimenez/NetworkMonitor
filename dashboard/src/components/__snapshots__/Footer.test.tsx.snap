// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > Footer matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="ant-layout-footer css-dev-only-do-not-override-14i19y2"
        style="text-align: center;"
      >
        © 
        2024
         Network Monitor. All rights reserved.
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="ant-layout-footer css-dev-only-do-not-override-14i19y2"
      style="text-align: center;"
    >
      © 
      2024
       Network Monitor. All rights reserved.
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer component > renders the footer with the correct year and text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="ant-layout-footer css-dev-only-do-not-override-14i19y2"
        style="text-align: center;"
      >
        © 
        2024
         Network Monitor. All rights reserved.
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="ant-layout-footer css-dev-only-do-not-override-14i19y2"
      style="text-align: center;"
    >
      © 
      2024
       Network Monitor. All rights reserved.
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
